container/dockertest/client.go

```diff
@@ -0,0 +1,268 @@
+package dockertest
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+	docker "github.com/docker/docker/client"
+)
+
+// Client satisfies docker.CommonAPIClient for mocking.
+type Client struct {
+	requests  *requests
+	responses *responses
+
+	// a placeholder for unimplemented methods.
+	docker.CommonAPIClient
+}
+
+// requests encapsulates request channels that keeps call arguments of Docker client methods.
+// Each call to a Docker client method piped to its request channel so multiple
+// call to a same method can be inspected by reading its channel multiple times.
+// Inspecting call arguments can be made by listening chan returned by LastX methods of *Testing.
+type requests struct {
+	negotiateAPIVersion   chan NegotiateAPIVersionRequest
+	swarmInit             chan SwarmInitRequest
+	networkCreate         chan NetworkCreateRequest
+	info                  chan InfoRequest
+	containerList         chan ContainerListRequest
+	containerInspect      chan ContainerInspectRequest
+	imageBuild            chan ImageBuildRequest
+	networkInspect        chan NetworkInspectRequest
+	networkRemove         chan NetworkRemoveRequest
+	taskList              chan TaskListRequest
+	serviceCreate         chan ServiceCreateRequest
+	serviceList           chan ServiceListRequest
+	serviceInspectWithRaw chan ServiceInspectWithRawRequest
+	serviceRemove         chan ServiceRemoveRequest
+	serviceLogs           chan ServiceLogsRequest
+}
+
+// responses encapsulates response channels that holds 'faked' return values of Docker client methods.
+// To fake a Docker client method's response send fake return values to it's
+// response channel. This can be made by calling ProvideX methods of *Testing.
+// We use channels here instead of setting values on the struct in case of a need for returning
+// conditional responses depending on request paramaters in future to deal with parallel calls made
+// to same client methods.
+type responses struct {
+	info                  chan infoResponse
+	imageBuild            chan imageBuildResponse
+	networkInspect        chan networkInspectResponse
+	networkCreate         chan networkCreateResponse
+	taskList              chan taskListResponse
+	serviceCreate         chan serviceCreateResponse
+	serviceList           chan serviceListResponse
+	serviceInspectWithRaw chan serviceInspectWithRawResponse
+	serviceRemove         chan serviceRemoveResponse
+	serviceLogs           chan serviceLogsResponse
+	contanerInspect       chan containerInspectResponse
```

Comment:
typo on `contanerInspect`. Should be `containerInspect`.
This typo is also at 3 other places.

Reply:
done